src/services/json-api-datastore.service.ts

```diff
@@ -161,18 +162,18 @@ export class JsonApiDatastore {
     }
 
     private extractQueryData<T extends JsonApiModel>(res: any, modelType: ModelType<T>, withMeta = false): T[] | JsonApiQueryData<T> {
-        let body: any = res.json();
+        let body: any = res;
         let models: T[] = [];
-        body.data.forEach((data: any) => {
-            let model: T = new modelType(this, data);
+        let model: T;
+        body.data.map((_data: any) => {
```

Comment:
can this bet written as
´models = body.data.map( ... )` 
and `return model` in the map function?